### PR TITLE
Prevent "document not defined" error in server side rendering.

### PR DESCRIPTION
**Pull request by wehriam**
See: https://github.com/facebook/react/issues/3620

### CODE REVIEW

File: lib/Select.js [hunk 1]
```diff
@@ -97,7 +98,10 @@ var Select = React.createClass({
 	componentWillMount: function componentWillMount() {
 		this._optionsCache = {};
 		this._optionsFilterString = '';
-		this.setState(this.getStateFromValue(this.props.value));
+		if(canUseDOM) {
+			this.setState(this.getStateFromValue(this.props.value));
```
Comment (sebmarkbage) [inline lib/Select.js]: Does this error actually fire here? If so, that is a bug in React 0.13.

The issue in facebook/react#3620 refers to callbacks to OTHER parent components from componentWillMount.

It should be fine to setState in componentWillMount, even on the server.

Perhaps one of the other callback fires or there's a bug in React 0.13 that can be fixed?

**Comment by sebmarkbage** [comment]
This is not the right feature detection since I should be able to renderToString even in a DOM environment. E.g. to send to an iframe, across workers, in a chrome extension etc.

I should also be able to render to a string on the server even if I have JSDOM enabled for some legacy integration points.

**Comment by xcatliu** [comment]
+1

**Comment by xcatliu** [comment]
You should not update `lib` files directly, maybe you should read [CONTRIBUTING.md](https://github.com/JedWatson/react-select/blob/master/CONTRIBUTING.md) first.

**Comment by agirton** [comment]
Closing as this is no longer applicable.

**Pull request closed without merging**